processing/src/main/java/org/apache/druid/query/filter/LikeDimFilter.java

```diff
@@ -364,6 +328,79 @@ public SuffixMatch getSuffixMatch()
       return suffixMatch;
     }
 
+    @VisibleForTesting
+    static class PatternDruidPredicateFactory implements DruidPredicateFactory
+    {
+      private final ExtractionFn extractionFn;
+      private final Pattern pattern;
+
+      PatternDruidPredicateFactory(ExtractionFn extractionFn, Pattern pattern)
+      {
+        this.extractionFn = extractionFn;
+        this.pattern = pattern;
+      }
+
+      @Override
+      public Predicate<String> makeStringPredicate()
+      {
+        if (extractionFn != null) {
+          return input -> matches(extractionFn.apply(input), pattern);
+        } else {
+          return input -> matches(input, pattern);
+        }
+      }
+
+      @Override
+      public DruidLongPredicate makeLongPredicate()
+      {
+        if (extractionFn != null) {
+          return input -> matches(extractionFn.apply(input), pattern);
+        } else {
+          return input -> matches(String.valueOf(input), pattern);
+        }
+      }
+
+      @Override
+      public DruidFloatPredicate makeFloatPredicate()
+      {
+        if (extractionFn != null) {
+          return input -> matches(extractionFn.apply(input), pattern);
+        } else {
+          return input -> matches(String.valueOf(input), pattern);
+        }
+      }
+
+      @Override
+      public DruidDoublePredicate makeDoublePredicate()
+      {
+        if (extractionFn != null) {
+          return input -> matches(extractionFn.apply(input), pattern);
+        } else {
+          return input -> matches(String.valueOf(input), pattern);
+        }
+      }
+
+      @Override
+      public boolean equals(Object o)
+      {
+        if (this == o) {
+          return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+          return false;
+        }
+        PatternDruidPredicateFactory that = (PatternDruidPredicateFactory) o;
+        return Objects.equals(extractionFn, that.extractionFn) &&
+               Objects.equals(pattern, that.pattern);
```

Comment:
`Pattern` doesn't implement `equals()`, you could use `Pattern.toString()` here and elsewhere

Reply:
good catch! Done

Reply:
I've added an inspection profile failure for this. It was being used in one other class